### PR TITLE
more explicit type for scale indexes / ids

### PR DESCRIPTION
**Pull request by nikodemus**
- enum PresetScales -> enum Scale
   - added explicit Scale::USER_SCALE
   - CANT_CHANGE_SCALE -> NO_SCALE
- Renaming for clarity:
  - Song::getCurrentPresetScale() -> Song::getCurrentScale()
  - Song::setScale() -> Song::setScaleNotes()
  - Song::setPresetScale() -> Song::setScale()
  - Song::userScale -> Song::userScaleNotes
- new toplevel functions dealing with enum Scale:
  - NoteSet::scaleId() -> getScale(), to avoid circular dep with the
    explicit type.
  - isUserScale()
- delete dead declaration: Song::addMajorDependentModeNotes()
- group Song methods working on scales a bit, and document them

### CODE REVIEW

File: src/deluge/model/clip/instrument_clip_minder.cpp [hunk 1]
```diff
@@ -525,14 +525,13 @@ void InstrumentClipMinder::drawActualNoteCode(int16_t noteCode) {
...
-	displayScaleName(newScale);
+	displayScaleName(currentSong->cycleThroughScales());
 }
 
 // Returns if the scale could be changed or not
-bool InstrumentClipMinder::setScale(int32_t newScale) {
-	int32_t result = currentSong->setPresetScale(newScale);
-	if (result == CANT_CHANGE_SCALE) {
+bool InstrumentClipMinder::setScale(Scale newScale) {
+	auto result = currentSong->setScale(newScale);
```
Comment (m-m-adams) [inline src/deluge/model/clip/instrument_clip_minder.cpp:533]: Could this be explicit? the return type isn't obvious at a glance

File: src/deluge/model/scale/preset_scales.h [hunk 2]
```diff
@@ -13,7 +13,7 @@
  * See OFFSET_6|5_NOTE_SCALE for how to deal with that.
  *
  * The actual definitions expand this macro, picking the thing they need:
- * - enum PresetScales
+ * - enum PresetScale
```
Comment (m-m-adams) [inline src/deluge/model/scale/preset_scales.h:16]: should this just be scale?

File: src/deluge/gui/ui/keyboard/column_controls/scale_mode.h [hunk 3]
```diff
@@ -35,8 +36,9 @@ class ScaleModeColumn : public ControlColumn {
 
 private:
 	int32_t currentScalePad = -1;
-	int32_t previousScale = -1;
-	uint8_t scaleModes[8] = {0, 1, 2, 3, 4, 5, 6, 7};
+	Scale previousScale = NO_SCALE;
+	Scale scaleModes[8] = {MAJOR_SCALE,  MINOR_SCALE,      DORIAN_SCALE,  PHRYGIAN_SCALE,
```
Comment (m-m-adams) [inline src/deluge/gui/ui/keyboard/column_controls/scale_mode.h:40]: Nit but could we remove scale from these enum names?

File: src/deluge/model/scale/preset_scales.h (hunk 2 above)
Comment (nikodemus) [reply src/deluge/model/scale/preset_scales.h:16]: Yes, it should. I must have updated from wrong branch! Traveling now, will fix later today or tomorrow.

Is force push ok once I find the right commits, or do you prefer a fixup commit?

Comment (nikodemus) [reply src/deluge/model/scale/preset_scales.h]: So, in principle yes, and I would like to do that, but we get some conflicts with `deluge::gui::ui::keyboard::controls::ChordModeChord` when doing that.

I'll look into making either one an enum class in a separate PR, if that's ok.

File: src/deluge/gui/ui/keyboard/column_controls/scale_mode.h (hunk 3 above)
Comment (m-m-adams) [reply src/deluge/gui/ui/keyboard/column_controls/scale_mode.h:40]: Yeah that's reasonable

**Review by m-m-adams — APPROVED** [review]

**Review by m-m-adams — APPROVED** [review]

**Pull request merged**